app/parse.py

```
@@ -14,10 +22,95 @@ class Product:
     rating: int
     num_of_reviews: int
 
+def scrape_products_from_page(driver: webdriver.Chrome, url: str, product_selector: str, use_more_button: bool = False) -> List[Product]:
+    driver.get(url)
+    wait = WebDriverWait(driver, 10)
+
+    try:
+        accept_btn = wait.until(EC.element_to_be_clickable((By.CSS_SELECTOR, ".accept-cookies-button")))
+        accept_btn.click()
+    except:
+        pass
```

The cookie accept handling uses `wait.until(EC.element_to_be_clickable(...))` inside a bare `try/except`. This will cause the full 10s wait on pages where the banner is not present and also hides other exceptions. To meet the requirement to handle the cookies banner without unnecessary delays, consider checking presence with a short explicit wait or `find_elements(...)` first, and catch specific exceptions (e.g., `TimeoutException`).